Tests/tests/test_FrontPanel.cpp

```diff
@@ -0,0 +1,486 @@
+        dispatcher->Activate(&service);
+    }
+
+    virtual ~FrontPanelInitializedEventTest() override
+    {
+        dispatcher->Deactivate();
+        dispatcher->Release();
+
+        PluginHost::IFactories::Assign(nullptr);
+    }
+};
+
+class FrontPanelInitializedEventDsTest : public FrontPanelInitializedEventTest {
+protected:
+    FrontPanelIndicatorImplMock frontPanelIndicatorImplStringMock;
+    FrontPanelIndicatorImplMock frontPanelIndicatorImplIntMock;
+    FrontPanelTextDisplayImplMock frontPanelTextDisplayImplStringMock;
+    FrontPanelTextDisplayImplMock frontPanelTextDisplayImplMock;
+    FrontPanelTextDisplayImplMock frontPanelTextDisplayImplIntMock;
+
+
+    FrontPanelInitializedEventDsTest()
+        : FrontPanelInitializedEventTest()
+    {
+        device::FrontPanelIndicator::getInstance("Power").impl = &frontPanelIndicatorImplStringMock;
+        device::FrontPanelIndicator::getInstance(0).impl = &frontPanelIndicatorImplIntMock;
+        device::FrontPanelTextDisplay::getInstance("Text").impl = &frontPanelTextDisplayImplStringMock;
+        device::FrontPanelTextDisplay::getInstance(0).impl = &frontPanelTextDisplayImplIntMock;
+        device::FrontPanelTextDisplay::getInstance().impl = &frontPanelTextDisplayImplMock;
+	
+    }
+
+    virtual ~FrontPanelInitializedEventDsTest() override
+    {
+        device::FrontPanelIndicator::getInstance("Power").impl = nullptr;
+        device::FrontPanelTextDisplay::getInstance("Text").impl = nullptr;
+        device::FrontPanelTextDisplay::getInstance().impl = nullptr;
+	device::FrontPanelTextDisplay::getInstance(0).impl = nullptr;
+        device::FrontPanelIndicator::getInstance(0).impl = nullptr;
+
+    }
+};
+
+TEST_F(FrontPanelTest, RegisteredMethods)
+{
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setBrightness")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("getBrightness")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("powerLedOn")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("powerLedOff")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setClockBrightness")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("getClockBrightness")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("getFrontPanelLights")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("getPreferences")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setPreferences")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setLED")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setBlink")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("set24HourClock")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("is24HourClock")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setClockTestPattern")));
+}
+
+TEST_F(FrontPanelDsTest, setBrightnessWIndex)
+{
+    ON_CALL(frontPanelIndicatorImplStringMock, setBrightness(::testing::_, ::testing::_))
+        .WillByDefault(::testing::Return());
```

Comment:
please check values in setBrightness

Reply:
@npoltorapavlo 
When you say check values, do you mean something like this:

```
ON_CALL(frontPanelIndicatorImplStringMock, setBrightness(::testing::_, ::testing::_))
        .WillByDefault(::testing::Invoke(
        [&](int brightness, bool persist){
        if(0<= brightness <= 100) //or whatever the check for valid parameters would be in this case
        {
                return;
        }
        else
         {
                //some kind of error response
         }
}));
```

Reply:
apparently should be EXPECT_CALL if intend to verify that the call is made
also EXPECT_EQ for important parameters
`EXPECT_EQ(brightness, 1);` ?